src/imagery/i.eodag/i.eodag.py

```diff
@@ -139,54 +166,110 @@ def get_bb(vector=None):
     }
 
 
-def download_by_id(query_id: str):
-    gs.verbose(
-        _(
-            "Searching for product ending with: {}".format(
-                query_id[-min(len(query_id), 8) :]
-            )
+def get_aoi(vector=None):
+    """Get the AOI for querying"""
+    # Handle empty AOI
+    if not vector:
+        return get_bb()
+
+    args = {}
+    args["input"] = vector
+
+    if gs.vector_info_topo(vector)["areas"] <= 0:
+        gs.fatal(_("No areas found in AOI map <{}>...").format(vector))
+    elif gs.vector_info_topo(vector)["areas"] > 1:
+        gs.warning(
+            _(
+                "More than one area found in AOI map <{}>. \
+                      Using only the first area..."
+            ).format(vector)
         )
-    )
-    product, count = dag.search(id=query_id)
-    if count != 1:
-        raise ParameterError("Product couldn't be uniquely identified")
-    if not product[0].properties["id"].startswith(query_id):
-        raise ParameterError("Product wasn't found")
-    gs.verbose(
-        _("Poduct ending with: {} is found.".format(query_id[-min(len(query_id), 8) :]))
-    )
-    gs.verbose(_("Downloading..."))
-    dag.download(product[0])
 
+    # are we in LatLong location?
+    s = gs.read_command("g.proj", flags="j")
+    kv = gs.parse_key_val(s)
```

Comment:
```suggestion
    kv = gs.parse_command("g.proj", flags="j")
```
This line is duplicated (see row 122/149). Maybe you could de-duplicate?